daemons/controld/controld_control.c

```diff
@@ -62,17 +62,23 @@ do_ha_control(long long action,
         if (pcmk_get_cluster_layer() == pcmk_cluster_layer_corosync) {
             registered = crm_connect_corosync(cluster);
         }
-#endif // SUPPORT_COROSYNC
 
         if (registered) {
+            const pcmk__node_status_t *node =
+                pcmk__get_node(cluster->nodeid, cluster->uname, NULL,
+                               pcmk__node_search_cluster_member);
+
             controld_election_init(cluster->uname);
             controld_globals.our_nodename = cluster->uname;
-            controld_globals.our_uuid = cluster->uuid;
-            if(cluster->uuid == NULL) {
+
+            free(controld_globals.our_uuid);
+            controld_globals.our_uuid = pcmk__corosync_uuid(node);
```

Comment:
Use a copy of `pcmk__cluster_node_uuid()` so this isn't corosync-specific and this function can stay library-private

Reply:
Good point